d2/src/test/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerTest.java

```diff
@@ -623,6 +626,113 @@ public void testGetClient() throws Exception
     loadBalancer.getClient(uriRequest, requestContextWithHint);
   }
 
+  /**
+   * Tests getClient() when with host override list specified in the request context.
+   * @throws Exception
+   */
+  @Test
+  public void testGetClientHostOverrideList() throws Exception
+  {
+    Map<String, LoadBalancerStrategyFactory<? extends LoadBalancerStrategy>> loadBalancerStrategyFactories =
+        new HashMap<>();
+    Map<String, TransportClientFactory> clientFactories = new HashMap<>();
+    List<String> prioritizedSchemes = new ArrayList<>();
+
+    MockStore<ServiceProperties> serviceRegistry = new MockStore<>();
+    MockStore<ClusterProperties> clusterRegistry = new MockStore<>();
+    MockStore<UriProperties> uriRegistry = new MockStore<>();
+
+    ScheduledExecutorService executorService = new SynchronousExecutorService();
+
+    loadBalancerStrategyFactories.put("degrader", new DegraderLoadBalancerStrategyFactoryV3());
+    clientFactories.put(PropertyKeys.HTTP_SCHEME, new DoNothingClientFactory());
+
+    SimpleLoadBalancerState state =
+        new SimpleLoadBalancerState(executorService,
+            uriRegistry,
+            clusterRegistry,
+            serviceRegistry,
+            clientFactories,
+            loadBalancerStrategyFactories);
+
+    SimpleLoadBalancer loadBalancer =
+        new SimpleLoadBalancer(state, 5, TimeUnit.SECONDS, _d2Executor);
+
+    FutureCallback<None> balancerCallback = new FutureCallback<None>();
+    loadBalancer.start(balancerCallback);
+    balancerCallback.get();
+
+    Map<Integer, PartitionData> partitionData = new HashMap<Integer, PartitionData>(1);
+    partitionData.put(DEFAULT_PARTITION_ID, new PartitionData(1d));
+    Map<URI, Map<Integer, PartitionData>> uriData = new HashMap<URI, Map<Integer, PartitionData>>(2);
+    uriData.put(URI.create("http://host1/path"), partitionData);
+
+    prioritizedSchemes.add(PropertyKeys.HTTP_SCHEME);
+
+    String cluster1 = "Cluster1";
+    String cluster2 = "Cluster2";
+    String service1 = "service1";
+    String service2 = "service2";
+
+    clusterRegistry.put(cluster1, new ClusterProperties(cluster1, Collections.emptyList(),
+        Collections.emptyMap(), new HashSet<>(), NullPartitionProperties.getInstance()));
+
+    serviceRegistry.put(service1, new ServiceProperties(service1,
+        cluster1,
+        "/service1Path",
+        Arrays.asList("degrader"),
+        Collections.<String,Object>emptyMap(),
+        null,
+        null,
+        prioritizedSchemes,
+        null));
+    uriRegistry.put(cluster1, new UriProperties(cluster1, uriData));
+
+    URI override = URI.create("http://override/path");
```

Comment:
Can we also have a test with override value with an ending slash URI.create("http://override/path/")

Reply:
Turned out the implementation is quite fragile that adding a suffix '\' will break the behavior. Given that the normal D2 path leverages the same concatenation. I recommend that we fix that all together in a different PR.

Reply:
sure